### DISCUSSION TITLE
iframe weather widget

### DISCUSSION THREAD

**Discussion by midair55** [VA3HDL/hamdashboard#42] [Show and tell]
I wanted a simple weather widget to display the current local conditions in one of the panels' as an iframe.  After a bit of trial and error, I found one that seems to work well with some modifications to the generated code for the widget.  This may come in handy for others that wish to do the same.  One caveat... It does NOT handle Daylight Savings Time and it does not appear there is a parameter to set the UTC offset.

https://www.weatherandradar.com/weather-widget

Search for your location and use the defaults for the other few settings.  The resulting HTML code is provided.  Copy ONLY the URL starting with "https://" and ending with the "temperatureUnit" value.  Do NOT use the other settings.  You will have to play with the, "region",  "systemOfMeasurement" and "temperatureUnit" settings in the URL if Celsius is desired since they default to, "US",  "imperial" and "fahrenheit".  It doesn't appear there is a way to set these in the web page's UI.

Configure a panel in the config.js using the "iframe|https://...."  such as the following for, 'Cape Canaveral, Florida':

[
    "",
    "iframe|https://api.wo-cloud.com/content/widget/?geoObjectKey=16031011&language=en&region=US&timeFormat=HH:mm&windUnit=mph&systemOfMeasurement=imperial&temperatureUnit=fahrenheit",
],

It's unclear how frequently the widget's data is updated so I configured the panel's refresh rate for 5 minutes (i.e. 300000).

My dashboard is:  https://qsl.net/n5ng/  if you wish to see the widget.

Clicking on the widget panel launches a new browser tab with the weather details.

Cheers!
Steve - N5NG

**Comment by VA3HDL** [comment]
Very nice!

I've used for my location:

"iframe|https://api.wo-cloud.com/content/widget/?geoObjectKey=38254799&language=en&region=CA&timeFormat=HH:mm&windUnit=kmh&systemOfMeasurement=metric&temperatureUnit=celsius",

works fine.

**Comment by midair55** [comment]
Sweet!  Glad to know that non-US locations are available and wind speeds in kilometers and temps in Celsius work!  One of the days the US may align with the world.....  but I'm not holding my breath!!

________________________________
